docker/release/cudaq.Dockerfile

```diff
@@ -88,9 +88,10 @@ RUN bash "$CUDA_QUANTUM_PATH/bin/migrate_assets.sh" "$CUDA_QUANTUM_PATH/assets"
     && rm "$CUDA_QUANTUM_PATH/bin/migrate_assets.sh"
 
 ENV PATH "${PATH}:$CUDA_QUANTUM_PATH/bin"
-ENV PYTHONPATH "${PYTHONPATH}:$CUDA_QUANTUM_PATH"
 ENV LD_LIBRARY_PATH="$LD_LIBRARY_PATH:$CUDA_QUANTUM_PATH/lib"
 ENV CPLUS_INCLUDE_PATH="$CPLUS_INCLUDE_PATH:$CUDA_QUANTUM_PATH/include"
+# Better alternative to setting the PYTHONPATH, since the PYTHONPATH is generally not preserved when running as sudo.
+RUN echo "$CUDA_QUANTUM_PATH" > /usr/local/lib/python3.10/dist-packages/cudaq.pth
```

Comment:
This is for the docker image, where we have a specific python version installed. We do not build it for/with different python versions. The path(s) in this file are added to the location that python searches for packages, see https://docs.python.org/3/library/site.html (curtesy of @1tnguyen). It replaces the line that added this location to the PYTHONPATH, since environment variables in general are not preserved when switching user with sudo (it can be configured to be preserved but this looks cleaner to me).